rfcs/table.md

```diff
@@ -0,0 +1,321 @@
+              <Td>Item</Td>
+              <Td>0.0640</Td>
+              <Td>Item</Td>
+            </Tr>
+            <Tr>
+              <Td>Item</Td>
+              <Td isNumerical={true}>1.0117</Td>
+              <Td>Item</Td>
+            </Tr>
+          </Tbody>
+       );
+     }
+    />
+}
+```
+
+### Advanced Usages
+
+#### Remote Data
+Paginate data from a remote backend.
+
+```javascript
+import * as React from 'react';
+import {Table} from 'baseui/table';
+import {Pagination} from 'baseui/pagination';
+
+const columns = {
+    {
+        title: 'Employee ID'
+        dataIndex: 'id'
+    },
+    {
+        title: 'First'
+        dataIndex: 'firstName'
+    },
+    {
+        title: 'Last'
+        dataIndex: 'lastName'
+    },
+    {
+        title: 'Full Name'
+        render: (rowData) => {
+            const {firstName, lastName} = rowData;
+            return (
+                <span>
+                    `${firstName} ${lastName}`
+                </span>
+            );
+        }
+    }
+};
+
+class TableWithPagination extends React.Component {
+    state = {
+        currentPage: 1,
+        pageSize: 10,
+        data: [],
+        numPages: 0
+    };
+    fetch = (pageNumber, pageSize) => {
+        // logic to fetch data based on page number
+        // `currentPage` and `data` and `numPages` will be updated here.
+    }
+    componentDidMount = () => {
+        const {currentPage, pageSize} = this.state;
+        this.fetch(currentPage, pageSize);
+    }
+    onPageChange = (nextPage, prevPage) => {
+        const {pageSize} = this.state;
+        this.fetch(nextPage, pageSize);
+    }
+    render = () => {
+         return (
+             <Table
+                columns={columns}
+                dataSource={this.state.data}
+             />
+             <Pagination currentPage={this.state.currentPage} numPages={this.state.numPages} onPageChange={this.onPageChange}/>
+         );
+    }
+}
+```
+
+
+## Exports
+
+* `Table`
+* `StatefulTable`
+* `Tbody`
+* `Thead`
+* `TheadTr`
+* `Td`
+* `Th`
+* `Tr`
+
+## `Table` API
+
+* `data:Array<T>` - Required
+  Description of prop
+* `columns: Column` - Required
+  Description of prop
+* `render: (dataSource, columns) => $ReactNode` - Optional
+    Custom render table body.
+* `overrides - Object:` - Optional
+
+* `overrides(continued):`
+  * `Tbody` - Body of the table. Corresponds.
+  * `Thead` - Container for the table header related elements.
+  * `TheadTr` - A header row. `Tr` for the `Thead` element.
+  * `Th` - A header cell. Corresponds with the `th` html element.
+  * `Td` - A data cell in the table. Corresponds with the `td` html element.
+  * `Tr` - A data row that contains `Td` components.
+
+
+## `Column` API
+One of the Table columns prop for describing the table's columns, Column has the same API.
+* `title: string` - Required. Title of the column to display in the header.
+* `dataIndex: string` - Optional. Attribute at which to index the row data.
```

Comment:
Sorry, I don't understand? What do you mean by "index the row data"?

Reply:
A row of data would be an object literal such as `{ firstName: 'John', lastName: 'Smith', id: 1}`.
`dataIndex` in this would be one of `firstName`, `lastName` or `id`.
For `index the row data`, I mean access the row object with a certain attribute.